src/frequenz/sdk/timeseries/_moving_window.py

```diff
@@ -138,6 +138,13 @@ def __init__(  # pylint: disable=too-many-arguments
         self._resampler_sender: Sender[Sample] | None = None
         self._resampler_task: asyncio.Task[None] | None = None
 
+        self._wait_for_num_samples: int = 0
+        """The number of samples to wait for before the wait_for_num_samples channels
+        sends out an event."""
```

Comment:
I recommend using nouns for non-boolean attributes:

```suggestion
        self._num_samples_to_wait_for: int = 0
        """The number of samples to wait for before triggering an event through the channel."""
```